src/targets/gpu/mlir.cpp

```diff
@@ -580,12 +599,93 @@ struct mlir_program
 
     std::string get_tune_params(bool xdlops) { return get_mlir_perf_for_conv(pp, xdlops); }
 
+    void tuning_table_create()
+    {
+        tuning_table               = mlirRockTuningTableCreate();
+        std::string tuning_db_path = string_value_of(MIGRAPHX_MLIR_TUNING_DB{});
+        if(!tuning_db_path.empty())
+        {
+            std::ifstream tuning_db_tsv(tuning_db_path);
+            if(tuning_db_tsv)
+            {
+                std::string arch, prob, perf, key;
+                std::getline(tuning_db_tsv, arch);
+                while(std::getline(tuning_db_tsv, arch, '\t'))
+                {
+                    std::getline(tuning_db_tsv, prob, '\t');
+                    std::getline(tuning_db_tsv, perf);
+                    key             = arch + "\t" + prob;
+                    char* key_cstr  = strdup(key.c_str());
+                    char* perf_cstr = strdup(perf.c_str());
+                    mlirRockTuningUpdateTable(tuning_table, key_cstr, perf_cstr, 1.0);
+                }
+                return;
+            }
+        }
+        std::cerr << "WARNING: MLIR tuning db not found. Please set MIGRAPHX_MLIR_TUNING_DB for "
+                     "optimal performance."
+                  << std::endl;
+    }
+
+    // This function appends to tuning cfg file that could be
+    // used with rocMLIR tuning scripts.
+    void dump_tuning_cfg(const char* probConfig)
+    {
+        std::string tuning_cfg_path = string_value_of(MIGRAPHX_MLIR_TUNING_CFG{});
+        if(!tuning_cfg_path.empty())
+        {
+            std::stringstream key(probConfig);
+            std::string prob;
+            int token_count = 0;
+            for(std::string s; getline(key, s, '\t');)
+            {
+                // first token is arch
+                // second one is the problem cfg
+                if(token_count == 1)
+                {
+                    prob = s;
+                }
+                if(token_count > 1)
+                {
+                    break;
```

Comment:
Actually, this probably should be `std::find_if` instead of `split_string` since you are only reading the first element, but either way would be fine though.

Reply:
done.